### ISSUE TITLE
AutoComplete Single Select Field State with Object Source

### ISSUE THREAD

**Issue by KerenChandran** [react-toolbox/react-toolbox#354]
Related to #252 

TLDR: Autocomplete for multiple={false} doesn't read from keys for selecting the entry on the list. Would be nice to fix it to only read from keys and always display the value associated to the key. The code I pasted below is a workaround to use the autocomplete with multiple={false}

Basically from the example below, I modified it to be a single select by adding the attribute multiple={false}. The initial state must be a key as the source code appears to be reading from the list of keys. 

When selecting a new value, the key associated to the value returns (which is good and should be expected). However, re-rendering the component causes the key to be displayed, as it seems that the code ignores looking up the value. This is a problem and so I thought it was a design implementation to go by values for single select autocomplete.

But wait! With a single select, setting the initial state to be the value associated to the key causes another error:
```javascript
TypeError: Cannot read property 'toLowerCase' of undefined
```

However, after the initial state being the key, setting the state of the component to be the value makes it work.

Please fix this or give me other suggestions. 

```javascript

const source = {'ES-es': 'Spain', 'TH-th': 'Thailand', 'EN-gb': 'England', 'EN-en': 'USA'};

class AutocompleteTest extends React.Component {
  state = {
    countries: 'ES-es'
  }

  handleChange = (value) => {
    this.setState({countries: source[value] });
  };

  render () {
  	console.log('state', this.state);
    return (
      <Autocomplete
        direction="down"
        label="Choose countries"
        name="countries"
        multiple= { false }
        onChange={this.handleChange}
        source={source}
        value={this.state.countries}
      />
    );
  }
}

return <AutocompleteTest />

```

**Comment by KerenChandran** [comment]
Actually the solution I posted works if that is the only component in the view. If the view is triggered to render by other components, then it will display the key.
